aptos-move/block-executor/src/scheduler.rs

```diff
@@ -3,51 +3,41 @@
 
 use aptos_infallible::Mutex;
 use crossbeam::utils::CachePadded;
+use parking_lot::{RwLock, RwLockUpgradableReadGuard};
 use std::{
-    cmp::min,
+    cmp::max,
     hint,
+    ops::DerefMut,
     sync::{
-        atomic::{AtomicBool, AtomicUsize, Ordering},
+        atomic::{AtomicBool, AtomicU64, AtomicUsize, Ordering},
         Arc, Condvar,
     },
 };
 
+const TXN_IDX_MASK: u64 = (1 << 32) - 1;
+
 // Type aliases.
 pub type TxnIndex = usize;
```

Comment:
should we change this to u32? it seems cleaner for the validator index use

Reply:
Since TxnIndex is also used in many other places other than validation_idx, I will keep the current one.

Reply:
I am merging the definitions in my diff (MVHashMap redefined and a TODO anyway), and I can try to redefine it there.